nest/04.finest.md

```diff
@@ -0,0 +1,192 @@
+# 4 finest
+As explained throughout this getting-started, `finest` is the command targeted towards experienced users.
+
+## 4.1 Key concepts
+Before using `finest`, you have to understand how it works on a deeper level.
+`finest` relies on 4 key concepts:
+* The cache
+* Transactions
+* The dependency graph
+* Groups
+
+### 4.1.1 The cache
+[//]: # (TODO: add link to the section 3.2)
+The cache was explained in [the section 3.2]().
+
+### 4.1.2 Transactions
+A transaction is an *action of `finest` on the filesystem*.
+A transaction can be, for example, removing or upgrading a package.
+
+### 4.1.3 The dependency graph
+The dependency graph is a directed graph representing the dependencies of packages.
+In this graph, each node is either **a group of packages**, either **a single package**.
+Each node has a requirement that is fulfilled by another node.
+A requirement can be fulfilled by **one** and only **one** package.
+A requirement is noted as `repository::category/package#[version|version_requirement]`.
+
+The dependency graph contains a root node named `@root`.
+Using `finest`, you can add or remove requirements from any node you want.
+You can't add a node without it having a parent node, so it will be `@root` by default.
+
+When you're done making your transactions with `finest`, you have to solve the dependency graph.
+`finest` has applied those transactions on a copy of your dependency graph, and when you solve it, it's actually *merging* that copy with the graph currently used by the OS.
+
+### 4.1.4 Groups
+A group is basically a group of packages.
```

Comment:
You should probably also say that all groups must have a name that starts with an `@`

Reply:
I said it in the infobox just below that paragraph. Should I add it in this paragraph instead?